redesign favorites dialog UI

adipose opened the pull request:
The saving of the favorites had several issues:

1. There was no apply button, making it unclear whether the favorites were persisted (to memory) or not
2. Closing the dialog silently failed to persist.  However, the list contents were actually maintained, so reopening the dialog appeared to indicate persistence.
3. Add favorites dialog can be opened at the same time, raising the question of how to merge favorites added to the organize dialog and those added via the "add" dialog.

This solves all issues by: 
1. Loading the persisted list whenever opening the organize dialog
2. adding an apply button
3. saving on apply or ok
4. offering to save when cancelling or "X"ing
5. When the organize dailog is open, handling anything added via the addfavorite dialog as if it were added to the organize dialog manually.
6. Add dialog works normally when organize is closed (it adds and persists immediately)

adipose committed:
redesign favorites dialog UI

adipose committed:
use lambda

adipose commented:
permanent download: https://mega.co.nz/#!ZQIhDZCa!LwVdDRD0aVeWY2AL9J5Pnm-QfQgvoiEIquD5hv6Rc8Q
github patch release: https://github.com/adipose/mpc-hc/releases/tag/patch545-favoritesdialogimprovements02

xLn2 commented:
Edit text and press ENTER keyboard key doesn't validate Apply button. Same happen with move up/down buttons. To be able to activate Apply button is click listview items.

Apply doesn't reset its current state (disabled/enabled) 2 click require to button as it should be. 1 apply click and close works but this visually distracts people.

Apply moves page to top.

adipose committed:
update apply state when making list changes

adipose committed:
do not reload listview on apply

adipose committed:
scroll to moved item in listview

adipose commented:
1. Edit text and press ENTER keyboard key doesn't validate Apply button.  - fixed
2. Same happen with move up/down buttons. - fixed
3. Apply doesn't reset its current state (disabled/enabled - fixed
4. Apply moves page to top. - fixed

permanent download: https://mega.co.nz/#!kcgREDrS!YXA6oGuaPkjeDZRI5RmLakvukxtgdeVwn4OfLLuskro
github patch release: https://github.com/adipose/mpc-hc/releases/tag/patch545-favoritesdialogimprovements03
